ui/src/explorer/AzureNode.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Subscription } from 'azure-arm-resource/lib/subscription/models';
+import { ServiceClientCredentials } from 'ms-rest';
+import { AzureEnvironment } from 'ms-rest-azure';
+import * as opn from 'opn';
+import { AzureExplorer, IAzureNode, IAzureParentNode, IAzureTreeItem } from '../../index';
+import { ArgumentError, NotImplementedError } from '../errors';
+
+export class AzureNode<T extends IAzureTreeItem = IAzureTreeItem> implements IAzureNode<T> {
+    public readonly treeItem: T;
+    public readonly parent: IAzureParentNodeInternal | undefined;
+    public constructor(parent: IAzureParentNodeInternal | undefined, treeItem: T) {
+        this.parent = parent;
+        this.treeItem = treeItem;
+    }
+
+    public get tenantId(): string {
+        if (this.parent) {
+            return this.parent.tenantId;
+        } else {
+            throw new ArgumentError(this);
+        }
+    }
+
+    public get subscription(): Subscription {
+        if (this.parent) {
+            return this.parent.subscription;
+        } else {
+            throw new ArgumentError(this);
+        }
+    }
+
+    public get credentials(): ServiceClientCredentials {
+        if (this.parent) {
+            return this.parent.credentials;
+        } else {
+            throw new ArgumentError(this);
+        }
+    }
+
+    public get environment(): AzureEnvironment {
+        if (this.parent) {
+            return this.parent.environment;
+        } else {
+            throw new ArgumentError(this);
+        }
+    }
+
+    public get explorer(): AzureExplorer {
+        if (this.parent) {
+            return this.parent.explorer;
+        } else {
+            throw new ArgumentError(this);
+        }
+    }
+
+    public refresh(): void {
+        this.explorer.refresh(this.parent, false);
+    }
+
+    public openInPortal(): void {
```

Comment:
Does this belong in the SiteWrapper in the azureappservices package?  This doesn't seem like a UI action to me.

Reply:
This applies for all types of azure resources, though, not just app services (cosmosdb account or storage accounts would have the same logic)